challenges/product/challenge-1.md

```diff
@@ -0,0 +1,156 @@
+# Product Challenge
+
+## Rationale
+
+This exercise has two goals:
+
+It helps us understand what to expect from you as a technical leader, and
+whether you have the technical aptitude to engage with highly technical
+customers. It gives you a feel for what it’s like to work at Teleport. The
+exercise simulates day-to-day work, helping you better understand the team and
+our hiring process.
+
+We believe this approach is not only more effective, but also more enjoyable
+than the whiteboard or quiz-style interviews common in the industry.
+
+We appreciate your time and look forward to hacking on this little project
+together.
+
+
+## Objective
+
+Create a design document and implement a service that will let users sign on
+through SSO IDP of of any kind and get a WireGuard public key set up on their
+computer to be authorized to access a WireGuard network.
+
+## Requirements
+
+Create automation using Terraform, helm charts, or docker-compose that sets up a
+demo environment with the following capabilities:
+
+- Determine implementation, security and scope requirements for the document.
+- Present the design to the interview team - a mix of technical and business audience, explain value proposition and answer technical questions to both.
+- Explain the business and technical value of the proposed solution.
```

Comment:
I found this section confusing because it talks about implementation details (Terraform, automation, etc) before mentioning the design document.

In fact, I'm not sure after reading this if we are expecting candidates to actually build the solution, of if we're just asking them to write the design document and present it to the interview team.

(Personally, I care much more about the quality of the design doc, the research the candidate did into this space and how compelling their presentation is than I care about their ability to write the code. That's not what we're hiring them for)

Reply:
We are asking them to build the end to end solution.